scripts/linters/pylint_extensions.py

```diff
@@ -2139,6 +2140,131 @@ def visit_classdef(self, node):
+                in the AST.
+
+        Returns:
+            bool. Whether URL_PATH_ARGS_SCHEMAS of a class is of dict type.
+        """
+        if 'URL_PATH_ARGS_SCHEMAS' not in node.locals:
+            return True
+
+        generator_object_for_value_of_schemas = (
+            node.locals['URL_PATH_ARGS_SCHEMAS'][0].assigned_stmts())
+
+        for value_of_schemas in generator_object_for_value_of_schemas:
+            if value_of_schemas.name != 'dict':
+                return False
+        return True
+
+    def check_handler_args_schemas_to_be_a_dict(self, node):
+        """Checks whether HANDLER_ARGS_SCHEMAS of a class is of dict type.
+
+        Args:
+            node: astroid.nodes.ClassDef. Node for a class definition
+                in the AST.
+
+        Returns:
+            bool. Whether HANDLER_ARGS_SCHEMAS of a class is of dict type.
+        """
+        if 'HANDLER_ARGS_SCHEMAS' not in node.locals:
+            return True
+
+        generator_object_for_value_of_schemas = (
+            node.locals['HANDLER_ARGS_SCHEMAS'][0].assigned_stmts())
+
+        for value_of_schemas in generator_object_for_value_of_schemas:
+            if value_of_schemas.name != 'dict':
+                return False
+        return True
+
+    def check_parent_class_is_basehandler(self, node):
+        """Checks whether the parent class of given class is BaseHandler.
+
+        Args:
+            node: astroid.nodes.ClassDef. Node for a class definition
+                in the AST.
+
+        Returns:
+            bool. Whether the parent class of given class is BaseHandler.
+        """
+        parent_class_is_basehandler = False
+        for ancestor_node in node.ancestors():
+            if ancestor_node.name == u'BaseHandler':
+                parent_class_is_basehandler = True
+                break
+        return parent_class_is_basehandler
+
+    def visit_classdef(self, node):
+        """Visit each class definition in controllers layer module and check
+        if it contains schema or not.
+
+        Args:
+            node: astroid.nodes.ClassDef. Node for a class definition
+                in the AST.
+        """
+        if not self.check_parent_class_is_basehandler(node):
+            return
+
+        if node.name in payload_validator.HANDLER_CLASS_NAMES_WITH_NO_SCHEMA:
+            return
+
+        if 'URL_PATH_ARGS_SCHEMAS' not in node.locals:
+            self.add_message(
+                'no-schema-for-url-path-elements', node=node, args=(node.name))
+        if 'HANDLER_ARGS_SCHEMAS' not in node.locals:
+            self.add_message(
+                'no-schema-for-handler-args', node=node, args=(node.name))
+        if not self.check_url_path_args_schemas_to_be_a_dict(node):
+            self.add_message(
+                'url-path-args-schemas-must-be-dict',
+                node=node, args=(node.name))
+        if not self.check_handler_args_schemas_to_be_a_dict(node):
```

Comment:
Ditto. Consider this as an else counterpart.

Reply:
Done